modules/core/core-elasticsearch7/src/main/java/com/enonic/xp/elasticsearch7/impl/ElasticsearchConfigResolver.java

```diff
@@ -0,0 +1,255 @@
+package com.enonic.xp.elasticsearch7.impl;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import com.enonic.xp.home.HomeDir;
+
+import static com.google.common.base.Strings.nullToEmpty;
+
+public final class ElasticsearchConfigResolver
+{
+
+    private static final String XP_HOME_PATTERN = "\\$\\{xp.home}";
```

Comment:
Hmmm. Placeholder substitution should happen much earlier in launcher. You should not need to solve it here